durmod fails to install with dev. Rcpp

Related to https://github.com/RcppCore/Rcpp/commit/cce9e618bb420e6c6a621ea56f1265ccb1c471cc.

```
kevinushey@Kevins-MBP:~/r/pkg
$ R CMD INSTALL --preclean durmod
* installing to library ‘/Users/kevinushey/Library/R/3.6/library’
* installing *source* package ‘durmod’ ...
** using staged installation
** libs
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Users/kevinushey/Library/R/3.6/library/Rcpp/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -I/usr/local/opt/libomp/include -Xpreprocessor -fopenmp    -fPIC  -g -O3 -pipe -Wall -pedantic -fstack-protector-all -c RcppExports.cpp -o RcppExports.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Users/kevinushey/Library/R/3.6/library/Rcpp/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -I/usr/local/opt/libomp/include -Xpreprocessor -fopenmp    -fPIC  -g -O3 -pipe -Wall -pedantic -fstack-protector-all -c cfunctions.cpp -o cfunctions.o
In file included from cfunctions.cpp:6:
In file included from /Users/kevinushey/Library/R/3.6/library/Rcpp/include/Rcpp.h:27:
In file included from /Users/kevinushey/Library/R/3.6/library/Rcpp/include/RcppCommon.h:128:
/Users/kevinushey/Library/R/3.6/library/Rcpp/include/Rcpp/exceptions.h:112:53: warning: all paths through this function will call
      itself [-Winfinite-recursion]
    inline void warning(const std::string& message) {        // #nocov start
                                                    ^
cfunctions.cpp:31:3: error: reference to 'debug' is ambiguous
  debug=dodebug;
  ^
cfunctions.cpp:28:13: note: candidate found by name lookup is 'debug'
static bool debug = false;
            ^
/Users/kevinushey/Library/R/3.6/library/Rcpp/include/Rcpp/macros/macros.h:28:11: note: candidate found by name lookup is
      'Rcpp::debug'
namespace debug {
          ^
1 warning and 1 error generated.
make: *** [cfunctions.o] Error 1
make: *** Waiting for unfinished jobs....
ERROR: compilation failed for package ‘durmod’
* removing ‘/Users/kevinushey/Library/R/3.6/library/durmod’
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2020-04-15  Kevin Ushey  <kevinushey@gmail.com>
+
+	* inst/include/Rcpp/String.h: Avoid 'debug' namespace collisions
+	* inst/include/Rcpp/macros/debug.h: Idem
+	* inst/include/Rcpp/macros/macros.h: Idem
+
 2020-04-15 Uwe Korn <mail@uwekorn.com>
 
 	* inst/include/Rcpp/DataFrame.h: Explicit call to scalar std::abs

--- a/inst/include/Rcpp/String.h
+++ b/inst/include/Rcpp/String.h
@@ -29,10 +29,10 @@
 
 #if RCPP_STRING_DEBUG_LEVEL > 0
     #define RCPP_STRING_DEBUG_FORMAT "%40s:%4d "
-    #define RCPP_STRING_DEBUG(MSG) Rprintf(RCPP_STRING_DEBUG_FORMAT "%s\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG);
-    #define RCPP_STRING_DEBUG_1(fmt, MSG) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG);
-    #define RCPP_STRING_DEBUG_2(fmt, M1, M2) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2);
-    #define RCPP_STRING_DEBUG_3(fmt, M1, M2, M3) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3);
+    #define RCPP_STRING_DEBUG(MSG) Rprintf(RCPP_STRING_DEBUG_FORMAT "%s\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG);
+    #define RCPP_STRING_DEBUG_1(fmt, MSG) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG);
+    #define RCPP_STRING_DEBUG_2(fmt, M1, M2) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2);
+    #define RCPP_STRING_DEBUG_3(fmt, M1, M2, M3) Rprintf(RCPP_STRING_DEBUG_FORMAT fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3);
 #else
     #define RCPP_STRING_DEBUG(MSG)
     #define RCPP_STRING_DEBUG_1(fmt, MSG)

--- a/inst/include/Rcpp/exceptions.h
+++ b/inst/include/Rcpp/exceptions.h
@@ -108,9 +108,9 @@ namespace Rcpp {
 
     // Variadic / code generated version of the warning and stop functions
     // can be found within the respective C++11 or C++98 exceptions.h
-    // included below
+    // included below.
     inline void warning(const std::string& message) {        // #nocov start
-        Rf_warning(message.c_str());
+        ::Rf_warning("%s", message.c_str());
     }                                                        // #nocov end
 
     inline void NORET stop(const std::string& message) {     // #nocov start

--- a/inst/include/Rcpp/macros/debug.h
+++ b/inst/include/Rcpp/macros/debug.h
@@ -33,12 +33,12 @@
 
 
 #if RCPP_DEBUG_LEVEL > 0
-    #define RCPP_DEBUG( MSG ) Rprintf( "%40s:%4d %s\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;
-    #define RCPP_DEBUG_1( fmt, MSG ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;
-    #define RCPP_DEBUG_2( fmt, M1, M2 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;
-    #define RCPP_DEBUG_3( fmt, M1, M2, M3 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3) ;
-    #define RCPP_DEBUG_4( fmt, M1, M2, M3, M4 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4) ;
-    #define RCPP_DEBUG_5( fmt, M1, M2, M3, M4, M5 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5) ;
+    #define RCPP_DEBUG( MSG ) Rprintf( "%40s:%4d %s\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;
+    #define RCPP_DEBUG_1( fmt, MSG ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;
+    #define RCPP_DEBUG_2( fmt, M1, M2 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;
+    #define RCPP_DEBUG_3( fmt, M1, M2, M3 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3) ;
+    #define RCPP_DEBUG_4( fmt, M1, M2, M3, M4 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4) ;
+    #define RCPP_DEBUG_5( fmt, M1, M2, M3, M4, M5 ) Rprintf( "%40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5) ;
 #else
     #define RCPP_DEBUG( MSG )
     #define RCPP_DEBUG_1( fmt, MSG )
@@ -52,49 +52,49 @@
     #define RCPP_DEBUG_MODULE( MSG ) {                                                                                             \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ){                                                                                                                 \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d %s\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d %s\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d %s\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;                                \
+            Rprintf( "[module () ] %40s:%4d %s\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;                                \
         }                                                                                                                          \
     }
     #define RCPP_DEBUG_MODULE_1( fmt, MSG ) {                                                                                      \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ){                                                                                                                 \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;                                \
+            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, MSG ) ;                                \
         }                                                                                                                          \
     }
     #define RCPP_DEBUG_MODULE_2( fmt, M1, M2 ) {                                                                                      \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ){                                                                                                                 \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;                                \
+            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2 ) ;                                \
         }                                                                                                                          \
     }
     #define RCPP_DEBUG_MODULE_3( fmt, M1, M2, M3 ) {                                                                                      \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ){                                                                                                                 \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3 ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3 ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3 ) ;                                \
+            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3 ) ;                                \
         }                                                                                                                          \
     }
     #define RCPP_DEBUG_MODULE_4( fmt, M1, M2, M3, M4 ) {                                                                                      \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ) {                                                                                                                \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4 ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4 ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4 ) ;                                \
+            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4 ) ;                                \
         }                                                                                                                          \
     }
     #define RCPP_DEBUG_MODULE_5( fmt, M1, M2, M3, M4, M5 ) {                                                                                      \
         Rcpp::Module * mod__ = getCurrentScope() ;                                                                                 \
         if( mod__ ){                                                                                                                 \
-            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5 ) ;\
+            Rprintf( "[module (%s) <%p> ] %40s:%4d " fmt "\n" , mod__->name.c_str(), mod__, ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5 ) ;\
         } else {                                                                                                                   \
-            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5 ) ;                                \
+            Rprintf( "[module () ] %40s:%4d " fmt "\n" , ::Rcpp::internal::debug::short_file_name(__FILE__).c_str(), __LINE__, M1, M2, M3, M4, M5 ) ;                                \
         }                                                                                                                          \
     }
 #else

--- a/inst/include/Rcpp/macros/macros.h
+++ b/inst/include/Rcpp/macros/macros.h
@@ -25,6 +25,7 @@
 #include <string>
 
 namespace Rcpp {
+namespace internal {
 namespace debug {
 
 inline std::string short_file_name(const char* file)
@@ -37,6 +38,7 @@ inline std::string short_file_name(const char* file)
 }
 
 } // namespace debug
+} // namespace internal
 } // namespace Rcpp
 
 #define RCPP_DECORATE(__FUN__) __FUN__##__rcpp__wrapper__

